### PR TITLE
Add test case for swagger DSS feature.

### PR DESCRIPTION
**Pull request by GDLMadushanka**

### CODE REVIEW

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    @AfterClass(alwaysRun = true)
+    public void destroy() throws Exception {
+        cleanup();
+    }
+
+    @Test(groups = {"wso2.dss"}, description = "Check swagger generation feature")
+    public void swaggerDataServiceWithResourcesTestCase() throws Exception {
+
+        String serviceEndPoint = "http://localhost:8480/services/ResourcesServiceTest?swagger.json";
+        HttpResponse response = this.getHttpResponse(serviceEndPoint);
```
Comment (Nirothipan) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java:81]: Shall we simple use HttpRequestUtil.doPost(new URL(url), PAYLOAD, headers); .. we don't need a method to get response as well.

**Comment by sdkottegoda** [comment]
Are we planning to address a combination of other resource methods, query params, path params and body params?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java (hunk 1 above)
Comment (GDLMadushanka) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java]: +1 fixed

**Comment by GDLMadushanka** [comment]
> Are we planning to address a combination of other resource methods, query params, path params and body params?

I addressed all the scenarios in the first test case. All methods, path/query parameter are covered. We check all the resources by comparing a single swagger JSON rather than checking one by one.

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    public void swaggerDataServiceWithResourcesTestCase() throws Exception {
+
+        String serviceEndPoint = "http://localhost:8480/services/ResourcesServiceTest?swagger.json";
+        HttpResponse response = HttpRequestUtil.doGet(serviceEndPoint, requestHeader);
+        String responseString = response.getData();
+        assertNotNull(responseString, "Failed to get the swagger response.");
+        JSONObject result = new JSONObject(responseString);
+        assertNotNull(result, "Response is null");
+        String pathDetails = result.get("paths").toString();
+        assertNotNull(result.get("paths"), "Swagger information should be available on all paths");
```
Comment (Nirothipan) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/swagger/SwaggerGenerationTestCase.java:89]: If this is null , we will get NPE in line 88 , isn't it ?

**Pull request merged**